src/_reducers/TradeReducer.js

```diff
@@ -37,7 +37,10 @@ export default (state = initialState, action) => {
             return initialState;
         }
         case DESTROY_TRADE: {
-            return state.delete(action.id);
+            if (state.size > 1) {
```

Comment:
Not sure I ever said that ;)
Definitely a reducer should not depend on api or outside thing, nor should call api or outside service.